plugin/build.gradle.kts

```
@@ -22,7 +23,12 @@ java.toolchain.languageVersion = JavaLanguageVersion.of(libs.versions.java.get()
 
 kotlin {
     abiValidation.enabled = true
-    compilerOptions.jvmDefault = JvmDefaultMode.NO_COMPATIBILITY
+    compilerOptions {
+        // https://docs.gradle.org/current/userguide/compatibility.html#kotlin
+        apiVersion = KotlinVersion.KOTLIN_2_0
+        languageVersion = apiVersion
+        jvmDefault = JvmDefaultMode.NO_COMPATIBILITY
+    }
```

```suggestion
    compilerOptions {
        apiVersion = KotlinVersion.KOTLIN_1_8
        jvmDefault = JvmDefaultMode.NO_COMPATIBILITY
    }
```
What about this setup instead? I tested it locally and it seems test are passing, even for `Gradle 7.3`.

I'd like to fully understand the impact of this, not my area of expertise

FYI: I can't downgrade `languageVersion` because then source code won't compile.